compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp

```diff
@@ -445,12 +445,32 @@ ComponentConfigurationImpl::CreateAndActivateComponentInstanceHelper(
   auto componentInstance = CreateComponentInstance();
   auto ctxt =
     std::make_shared<ComponentContextImpl>(shared_from_this(), bundle);
+  /*
+   * Failing to construct the servce object is an unrecoverable 
+   * failure which will cause the service configuration to not
+   * be active.
+   */
   try {
-    componentInstance->CreateInstanceAndBindReferences(ctxt);
+    componentInstance->CreateInstance(ctxt);
   } catch (const std::exception&) {
     logger->Log(
       cppmicroservices::logservice::SeverityLevel::LOG_ERROR,
-      "Exception while creating component instance and binding references.",
+      "Exception while creating component instance.",
+      std::current_exception());
+    throw;
```

Comment:
Just wondering if it's okay to leave the ComponentConfigurationImpl object in the state it's in. Maybe this component should be disabled since it can't be constructed?

Reply:
The OSGi spec doesn't make any assertion about moving the state to disabled. The state will remain in satisfied and never move to active.